### PR TITLE
Bug 1316278 - Implement Dockerflow health checks

### PR DESCRIPTION
**Pull request by jezdez**
Many thanks to @mythmon for the inspiration in the Normandy code.
This also removes the need for Newrelic deploys to Heroku.

### CODE REVIEW

**Comment by jezdez** [comment]
@maurodoglio r?

**Comment by jezdez** [comment]
This kind of drops the coverage a bunch, guess we should add some tests to this :-/

File: bin/pre_deploy [hunk 1]
```diff
@@ -16,23 +16,4 @@ fi
...
-# buildpack compile, so that in the case of deploy failures it's up to date.
-OLD_REVISION="$(curl --silent --show-error --fail --retry 5 --retry-max-time 15 $SITE_URL/revision.txt)"
-CHANGELOG="https://github.com/mozilla/telemetry-analysis-service/compare/$OLD_REVISION...$HEROKU_SLUG_COMMIT"
-# The author of the deploy isn't currently available to us. Have filed:
-# https://help.heroku.com/tickets/343783
-USER="Heroku"
-# Report the deploy to New Relic using their Python agent. In addition to
-# the passed arguments, record-deploy references the environment variables
-# `NEW_RELIC_APP_NAME` and `NEW_RELIC_API_KEY`.
-newrelic-admin record-deploy "$NEW_RELIC_CONFIG_FILE" \
```
Comment (maurodoglio) [inline bin/pre_deploy]: Don't we still need to send a record-deploy to newrelic?

**Comment by mythmon** [comment]
@jezdez This is now the third (that I know of) implementation of __heartbeat__, __version__, and __lbheartbeat__. What would you think about making this into a common library we could all use to help implement Dockerflow?

**Comment by jezdez** [comment]
@mythmon yeah, that'd be useful, I'd be happy to help out. Maybe ship a bunch of those checks out of the box and enable them depending on a setting? Would be interesting to have some of the other Dockerflow requirements covered as well, like mozlog and a set of Dockerfiles with Python 2 and 3?

File: bin/pre_deploy (hunk 1 above)
Comment (jezdez) [inline bin/pre_deploy]: @robotblake said the Jenkins deploy task takes care of that in Dockerflow

Comment (maurodoglio) [inline bin/pre_deploy]: 🆒

**Comment by jezdez** [comment]
@mythmon I've kickstarted the idea in https://github.com/mozilla-services/python-dockerflow and added you, @maurodoglio and @Osmose to the project in case you're interested to collaborate

**Pull request merged**